src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,438 @@
+                    for p in probs
+                ],
+                dim="ess_dim",
+            )
+            # print(f"\n ess_y_dataset = {ess_y_dataset}")
+
+            # converting xdata into a xr dataarray
+            xdata_da = xr.DataArray(xdata, dims="ess_dim")
+            # print(f"\n xdata_da ={xdata_da}")
+
+            # broadcasting xdata_da to match shape of each variable in ess_y_dataset and
+            # creating a new dataset from dict of broadcasted xdata
+            xdata_dataset = xr.Dataset(
+                {var_name: xdata_da.broadcast_like(da) for var_name, da in ess_y_dataset.items()}
+            )
+            # print(f"\n xdata_dataset = {xdata_dataset}")
+
+            # concatenating xdata_dataset and ess_y_dataset along plot_axis
+            ess_dataset = xr.concat([xdata_dataset, ess_y_dataset], dim="plot_axis").assign_coords(
+                plot_axis=["x", "y"]
+            )
+            print(f"\n distribution = {distribution!r}")
+            print(f"\n ess_dataset = {ess_dataset!r}")
+
+            # step 4
+            # if "color" not in ess_aes:
+            #    ess_kwargs.setdefault("color", "gray")
+
+            # step 5
+            plot_collection.map(
+                scatter_xy, "local", data=ess_dataset, ignore_aes=ess_ignore, **ess_kwargs
+            )
+
+        if kind == "quantile":
+            probs = np.linspace(1 / n_points, 1 - 1 / n_points, n_points)
+            xdata = probs
+            ylabel = "{} for quantiles"
+
+            # step 3
+            ess_y_dataset = xr.concat(
+                [
+                    distribution.azstats.ess(
+                        dims=ess_dims,
+                        method="quantile",
+                        relative=relative,
+                        prob=p,
+                        **stats_kwargs.get("ess", {}),
+                    )
+                    for p in probs
+                ],
+                dim="ess_dim",
+            )
+
+            # converting xdata into an xr datarray
+            xdata_da = xr.DataArray(xdata, dims="ess_dim")
+
+            # broadcasting xdata_da to match shape of each variable in ess_y_dataset and
+            # creating a new dataset from dict of broadcasted xdata
+            xdata_dataset = xr.Dataset(
+                {var_name: xdata_da.broadcast_like(da) for var_name, da in ess_y_dataset.items()}
+            )
+
+            # concatenating xdata_dataset and ess_y_dataset along plot_axis
+            ess_dataset = xr.concat([xdata_dataset, ess_y_dataset], dim="plot_axis").assign_coords(
+                plot_axis=["x", "y"]
+            )
+            print(f"\n ess_dataset = {ess_dataset!r}")
+
+            # step 4
+            # if "color" not in ess_aes:
+            #    ess_kwargs.setdefault("color", "gray")
+
+            # step 5
+            plot_collection.map(
+                scatter_xy, "quantile", data=ess_dataset, ignore_aes=ess_ignore, **ess_kwargs
+            )
```

Comment:
If all this is moved outside the if. To get the same exact behaviour using `.map(scatter_xy, kind, ...)` would be needed, but from #66 I think it should instead be `.map(scatter_xy, "ess", ...)`